source/scripts/app.js

```diff
@@ -17,6 +17,60 @@ function createRecipeCard() {
   return recipeCard;
 }
 
+/**
+ * This function toggles whether the explore page will display recipes based on a filter or
+ * by random.
+ */
+function toggleExplorePageType() {
+  "use strict";
+  let shadow = document.querySelector("explore-page").shadowRoot;
+  let topLevel = shadow.getElementById("explore-top-level");
+  let loadButton = shadow.getElementById("load-button");
+  topLevel.classList.toggle("type-explore");
+
+  if (topLevel.classList.contains("type-explore")) {
+    loadButton.textContent = "Explore More";
+  } else {
+    loadButton.textContent = "Explore Recipes";
+  }
+}
+
+/**
+ * Populates new recipes in the Explore page by retrieving new recipes from
+ * Spoonacular
+ * @function populateExplorePage
+ */
+async function populateExplorePage(filtersObj) {
+  "use strict";
+  let shadow = document.querySelector("explore-page").shadowRoot;
+  let topLevel = shadow.getElementById("explore-top-level");
+
+  if (topLevel.classList.contains("type-explore")) {
+    let recipes = await spoonacular.getRandomRecipes(EXPLORE_PAGE_NUM_RESULTS);
+    shadow.getElementById("no-results-text").classList.add("make-invisible");
+    let recipeCards = shadow.getElementById("recipe-cards-section").children;
+
+    for (let i = 0; i < recipes.length; ++i) {
+      recipeCards[i].classList.remove("make-invisible");
+      let shadow = recipeCards[i].shadowRoot;
+      shadow.getElementById("recipe-id").textContent = recipes[i].id;
+      shadow.getElementById("recipe-card-title").textContent = recipes[i].title;
+      shadow.getElementById("recipe-card-image").src = recipes[i].image;
+    }
+  } else {
+    let recipes = await spoonacular.getRecipes(filtersObj);
+    let recipeCards = shadow.getElementById("recipe-cards-section").children;
+
+    for (let i = 0; i < recipes.length; i++) {
+      let cardShadow = recipeCards[i].shadowRoot;
+      cardShadow.getElementById("recipe-id").textContent = recipes[i].id;
+      cardShadow.getElementById("recipe-card-title").textContent =
+        recipes[i].title;
+      cardShadow.getElementById("recipe-card-image").src = recipes[i].image;
+    }
+  }
```

Comment:
This can be refactored to make it more compact
```suggestion
  let recipes = {};
  if (topLevel.classList.contains("type-explore")) {
    recipes = await spoonacular.getRandomRecipes(EXPLORE_PAGE_NUM_RESULTS);
  } else {
    recipes = await spoonacular.getRecipes(filtersObj);
  }
  shadow.getElementById("no-results-text").classList.add("make-invisible");
  let recipeCards = shadow.getElementById("recipe-cards-section").children;

  for (let i = 0; i < recipes.length; ++i) {
    recipeCards[i].classList.remove("make-invisible");
    let cardShadow = recipeCards[i].shadowRoot;
    cardShadow.getElementById("recipe-id").textContent = recipes[i].id;
    cardShadow.getElementById("recipe-card-title").textContent = recipes[i].title;
    cardShadow.getElementById("recipe-card-image").src = recipes[i].image;
  }
```